Managers/LuaMan.cpp

```diff
@@ -1,2829 +1,474 @@
+			g_ConsoleMan.PrintString("Error: Can't open file, maximum number of files already open.");
+			return -1;
+		}
 
-	// Do not open paths with '..'
-	if (fullPath.find(dotString) != string::npos)
-		return -1;
+		std::string fullPath = System::GetWorkingDirectory() + fileName;
 
-	// Do not open paths that aren't written correctly
-	if (!System::PathExistsCaseSensitive(std::filesystem::path(filename).lexically_normal().generic_string()))
-		return -1;
+		// Do not open paths with '..'
+		if (fullPath.find("..") != std::string::npos) {
+			return -1;
+		}
+		// Do not open paths that aren't written correctly
+		if (!System::PathExistsCaseSensitive(std::filesystem::path(fileName).lexically_normal().generic_string())) {
+			return -1;
+		}
+		// Allow to edit files only inside .rte folders
+		if (fullPath.find(System::GetModulePackageExtension()) == std::string::npos) {
+			return -1;
+		}
 
-	// Allow to edit files only inside .rte folders
-	if (fullPath.find(rteString) == string::npos)
+		if (FILE * file = fopen(fullPath.c_str(), accessMode.c_str())) {
+			m_OpenedFiles.at(fileIndex) = file;
+			return fileIndex;
+		}
+		g_ConsoleMan.PrintString("Error: Failed to open file " + fullPath);
 		return -1;
-
-	// Open file and save handle
-	FILE * f = fopen(fullPath.c_str(), mode.c_str());
-	if (f)
-	{
-		m_Files[fl] = f;
-		return fl;
 	}
 
-#ifdef _DEBUG
-	g_ConsoleMan.PrintString("Error: Can't open file. " + fullPath);
-#endif
-	return -1;
-}
-
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::FileClose(int file)
-{
-	if (file > -1 && file < MAX_OPEN_FILES && m_Files[file])
-	{
-		fclose(m_Files[file]);
-		m_Files[file] = 0;
+	void LuaMan::FileClose(int fileIndex) {
+		if (fileIndex > -1 && fileIndex < c_MaxOpenFiles && m_OpenedFiles.at(fileIndex)) {
+			fclose(m_OpenedFiles.at(fileIndex));
+			m_OpenedFiles.at(fileIndex) = nullptr;
+		}
 	}
-}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::FileCloseAll()
-{
-	for (int file = 0 ; file < MAX_OPEN_FILES ; ++file)
-		FileClose(file);
-}
+	void LuaMan::FileCloseAll() {
+		for (int file = 0; file < c_MaxOpenFiles; ++file) {
+			FileClose(file);
+		}
+	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-std::string LuaMan::FileReadLine(int file)
-{
-	if (file > -1 && file < MAX_OPEN_FILES && m_Files[file])
-	{
-		char buf[4096];
-		fgets(buf, 4095, m_Files[file]);
-		return (std::string(buf));
+	std::string LuaMan::FileReadLine(int fileIndex) {
+		if (fileIndex > -1 && fileIndex < c_MaxOpenFiles && m_OpenedFiles.at(fileIndex)) {
+			char buf[4096];
+			fgets(buf, 4095, m_OpenedFiles.at(fileIndex));
+			return (std::string(buf));
+		}
+		g_ConsoleMan.PrintString("Error: Tried to read a closed file, or read past EOF.");
```

Comment:
Regarding the console man error, does this check EOF, or is this error kinda wrong?
Also, rather than `a closed file` I guess it'd be more correct to say `an invalid or closed file`. Though maybe lua users don't need that extra info.